### PR TITLE
chore: export RouteGenericInterface

### PR DESCRIPTION
**Pull request by ChrisCrewdson**
This is a very small change just to re-export RouteGenericInterface which is needed by typescript 4.8.x for implementations of RouteHandlerMethod
I created an issue for this https://github.com/fastify/fastify/issues/4235 that will be solved by this PR.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Could you add a unit test for this?

**Comment by ChrisCrewdson** [comment]
I am not sure if I can, since this is just about being able to access the type but I can try.

**Comment by mcollina** [comment]
We use tsd for our types tests.

**Comment by ChrisCrewdson** [comment]
Since this doesn't actually add a type, just re-exports it, I can't figure out how this fits as a test. The existing type just needs to be exposed.

**Comment by mcollina** [comment]
import it in https://github.com/fastify/fastify/blob/main/test/types/fastify.test-d.ts#L12 add an `expectType`.

**Comment by ChrisCrewdson** [comment]
Hopefully this is what you are looking for. This is the simplest test case since all properties from the interface are optional.

File: test/types/fastify.test-d.ts [hunk 1]
```diff
@@ -232,3 +233,6 @@ const ajvErrorObject: AjvErrorObject = {
   message: ''
 }
 expectAssignable<ValidationResult>(ajvErrorObject)
+
+const routeGeneric: RouteGenericInterface = {}
+expectType<RouteGenericInterface>(routeGeneric)
```
Comment (Uzlopak) [inline test/types/fastify.test-d.ts:238]: Well, that is quite circular.

With this you test atleast the shape.
```suggestion
expectType<unknown>(routeGeneric.Body)
expectType<unknown>(routeGeneric.Headers)
expectType<unknown>(routeGeneric.Params)
expectType<unknown>(routeGeneric.Querystring)
expectType<unknown>(routeGeneric.Reply)
```

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**